GitTools/GitReleaseManager
# Add support for GitLab

## Description

This PR adds support to allow creation, and maintenance of, releases on GitLab.  This has been made possible using the NGItLab library.  There are some items that have been left unimplemented, see the commit messages for more details.

## Related Issue

Fixes #146 

## Motivation and Context

I work on some projects that use GitLab for private work, and I want the ability to create release notes in the same way as done on public projects on GitHub.

## How Has This Been Tested?

1. Create both a public and private repository on GitLab
2. Get token to allow maintenance of these projects
3. Create a milestone and some issues using standard GRM labels
4. Run the `create` command to create release from the milestone - this will create an "upcoming release"
5. Run the `discard` command to delete the release
6. Run the `create` command again
7. Run the `publish` command to update the release to actually being releases
8. Run the `close` command to close the milestone
9. Run the `open` command to open the milestone
10. Run the `export` command to grab a markdown file version of the release that was created

The above are some sample things to do to verify things are working as expected, but various other tests have also been completed.  Let me know if you run into any problems with anything.

## Screenshots (if appropriate):

N#A

## Checklist:
